modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/extension/v1_0/ObjectRelationshipExtensionProvider.java

```diff
@@ -222,15 +287,58 @@ private ObjectDefinition _getRelatedObjectDefinition(
 			ObjectRelationship objectRelationship)
 		throws Exception {
 
-		long objectDefinitionId1 = objectRelationship.getObjectDefinitionId1();
+		long relatedObjectDefinitionId;
 
-		if (objectDefinitionId1 != objectDefinition.getObjectDefinitionId()) {
-			return _objectDefinitionLocalService.getObjectDefinition(
-				objectRelationship.getObjectDefinitionId1());
+		if (objectRelationship.getObjectDefinitionId1() ==
+				objectDefinition.getObjectDefinitionId()) {
+
+			relatedObjectDefinitionId =
+				objectRelationship.getObjectDefinitionId2();
+		}
+		else {
+			relatedObjectDefinitionId =
+				objectRelationship.getObjectDefinitionId1();
 		}
 
-		return _objectDefinitionLocalService.getObjectDefinition(
-			objectRelationship.getObjectDefinitionId2());
+		ObjectDefinition relatedObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				relatedObjectDefinitionId);
+
+		if (!relatedObjectDefinition.isActive()) {
+			throw new BadRequestException(
+				"ObjectDefinition " +
+					relatedObjectDefinition.getObjectDefinitionId() +
+						" must be activated");
+		}
+
+		if (relatedObjectDefinition.isSystem()) {
```

Comment:
Same here with the exception, maybe a more specific one. Is this possible to be a system object definition in here? As I guess it should have been filtered before invoking this method. In that case, maybe it should be a server error